### DISCUSSION TITLE
How to apply socket for video in MERN stack project?

### DISCUSSION THREAD

**Discussion by popstar7** [community/community#131183] [Programming Help] [answered]
### Select Topic Area

Question

### Body

Now I am building MERN stack project.
I don't understand how to insert socket in my project.
There are chatting rooms and watching video rooms.
Please help me.

**Comment by noahstoner** [comment, marked as answer]
You can add chatting room with socket.io library.
But video room can't be add with socket.io.
For this, there is stream.
Let's try it.
Wish you success.

**Reply by popstar7** [reply]
Thank you for your care.
I will try.
For your support, I am marking your answer.

**Reply by noahstoner** [reply]
Wish you success

**Comment by Er-Sadiq** [comment]
You Can Follow These Steps to Integrate Socket.io in a MERN Stack Project
Set Up Your Project:

Install socket.io on the server side.
Install socket.io-client on the client side.
Set Up Socket.io on the Server:

Create or update your server file to integrate socket.io.
Set up event listeners for connection, joining rooms, sending messages, and disconnection.
Set Up Socket.io on the Client:

Initialize socket.io-client in your main React component.
Set up event listeners for receiving messages and emitting messages.
